### ISSUE TITLE
SSH ControlPersist not working in docker container on CoreOS host, works on Ubuntu host

### ISSUE THREAD

**Issue by tleyden** [coreos/bugs#376]
When I use ssh ControlPersist within a docker container on a CoreOS host, it's not working.  Whereas if I do it from a docker container from the same image on an Ubuntu host it works.

**Steps to reproduce:** (identical steps on each type of host - CoreOS/Ubuntu)
- startup a centos7 host on AWS using ami-96a818fe, which has a default user of centos
- `docker run -ti tleyden5iwx/perfcluster-aws /bin/bash`
- Inside the docker container
  - `ssh-keygen` and leave all default values
  - Add public key to remote centos7 host in authorized_keys
  - `ssh -vvvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath=/tmp/%r@%h:%p centos@$centos7-host-ip`

[Actual result (CoreOS host)](https://gist.github.com/tleyden/debd112577e1f1ccab78)

```
Control socket connect(/tmp/centos@54.237.94.9:22): Connection refused
Failed to connect to new control master
```

[Expected result (Ubuntu host)](https://gist.github.com/tleyden/9ab71ada33491f0836ff)

(no errors)

**Comment by varsy** [comment]
+1, ansible playbooks fails with the following error: 
```
SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh
```

**Comment by crawford** [comment]
@tleyden are you using the btrfs driver or overlayfs with Docker? If you are using overlayfs, it would be worth trying to reproduce the problem with btrfs. You can create a file-backed btrfs volume as described [here](https://coreos.com/docs/cluster-management/setup/mounting-storage/#creating-and-mounting-a-btrfs-volume-file).

**Comment by tleyden** [comment]
@crawford do you mean on the CoreOS host (can repro) or on the Ubuntu host (cannot repro)?

Also, how can I tell which driver (btrfs vs overlayfs) I am using?

**Comment by crawford** [comment]
Sorry, on the CoreOS host. You can use `docker info` to figure out the storage driver.

**Comment by meAmidos** [comment]
+1, the same problem

**Comment by mlosev** [comment]
I ran the same issue with docker 1.6.2 + overlay storage driver
When I mounted directory from host and created socket there - it start working
Try to create control sockets on bind-mounted folder

**Comment by philips** [comment]
@vcaputo is this the overlayfs unix socket handling?

**Comment by juris** [comment]
+1 CoreOS 899.17.0 with Docker 1.9.1

**Comment by arthurtsang** [comment]
https://groups.google.com/forum/#!topic/ansible-project/y8ohlv_dRi4

putting the control path to tmpfs resolves the issue.

**Comment by euank** [comment]
According to https://github.com/moby/moby/issues/12080#issuecomment-226435505 this should be fixed since kernel 4.7.

All the Container Linux channels are on 4.11+, so this bug should be fixed.
